staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -229,7 +229,9 @@ func (f *SimpleRestOptionsFactory) GetRESTOptions(resource schema.GroupResource)
 		if !ok {
 			cacheSize = f.Options.DefaultWatchCacheSize
 		}
-		ret.Decorator = genericregistry.StorageWithCacher(cacheSize)
+		if cacheSize >= 0 {
+			ret.Decorator = genericregistry.StorageWithCacher(cacheSize)
```

Comment:
StorageWithCacher already skips caching if cacheSize is 0:

https://github.com/kubernetes/kubernetes/blob/6de09f69cb08a6dbde9912c1b9c87326fb11380f/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go#L45-L48

suggest changing that to `if capacity <= 0` and not adding a cacheSize check here and in StorageFactoryRestOptionsFactory#GetRESTOptions

Reply:
@liggitt Indeed if `StorageWithCacher` can return a `rawStorage` depending on the capacity this could be easier. we could even simplify the decorator logic we're using here.

Reply:
Given that it's already doing that, we should simplify it.
@liggitt - great catch